file-formats/cdbo/cdbo/type/zif_aff_cdbo_v1.intf.abap

```diff
@@ -0,0 +1,85 @@
+INTERFACE zif_aff_cdbo_v1
+  PUBLIC.
+  "! <p class="shorttext">Object Type</p>
+  "! Object type
+  "! $values {@link zif_aff_cdbo_v1.data:co_object_type}
+  TYPES ty_object_type TYPE c LENGTH 2.
+
+  CONSTANTS:
+    "! <p class="shorttext">Object Type</p>
+    "! Object type
+    BEGIN OF co_object_type,
+      "! <p class="shorttext">Tables</p>
+      "! Tables
+      tables    TYPE ty_object_type VALUE '01',
+      "! <p class="shorttext">CDS Views</p>
+      "! CDS views
+      cds_views TYPE ty_object_type VALUE '02',
+    END OF co_object_type.
+
+  "! <p class="shorttext">Field Nature</p>
+  "! Nature of the field
+  "! $values {@link zif_aff_cdbo_v1.data:co_field_nature}
+  TYPES ty_field_nature TYPE c LENGTH 2.
+
+  CONSTANTS:
+    "! <p class="shorttext">Field Nature</p>
+    "! Nature of the field
+    BEGIN OF co_field_nature,
+      "! <p class="shorttext">General fields that do not contain sensitive personal info.</p>
+      "! General fields that do not contain sensitive personal info.
+      general TYPE ty_field_nature VALUE 'GE',
+      "! <p class="shorttext">Sensitive personal information fields including key fields</p>
+      "! Sensitive personal information fields including key fields
+      sensitive    TYPE ty_field_nature VALUE 'SP',
+      "! <p class="shorttext">Non Business Information</p>
+      "! Non business information
+      non_business_information TYPE ty_field_nature VALUE 'NB',
+    END OF co_field_nature.
+
+  TYPES:
+    "! <p class="shorttext">CDBO Details</p>
+    "! CDBO details
+    BEGIN OF ty_field,
+      "! <p class="shorttext">Field Name</p>
+      "! Field name
+      "! $required
+      field_name TYPE c LENGTH 30,
+      "! <p class="shorttext">Field Nature</p>
+      "! Field nature
+      "! $required
+      field_nature TYPE ty_field_nature,
```

Comment:
I suggest to remove the prefix `field_`. You are already in the structure/array for "fields".

```suggestion
      "! <p class="shorttext">Name</p>
      "! Field name
      "! $required
      name TYPE c LENGTH 30,
      "! <p class="shorttext">Nature</p>
      "! Field nature
      "! $required
      nature TYPE ty_field_nature,
```